### PR TITLE
configured interceptor

### PR DESCRIPTION
**Pull request by mmartyniuk**

### CODE REVIEW

File: src/client/app/header/app.header.controller.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            function(){
+                if($sessionStorage.token){
+                    return vm.loggedIn = true;
+                }
+            });
+        vm.logOut = function(){
+            $sessionStorage.$reset();
+            $state.go($state.current, {}, {reload: true});
+            vm.loggedIn = false;
+        }
```
Comment (standemchuk) [inline src/client/app/header/app.header.controller.js]: you should also make a request to backend here, to revoke the token

File: src/client/app/needs/new_need_register.controller.js [hunk 2]
```diff
@@ -48,7 +50,11 @@
         };
 
         function activate() {
-            vm.regions = CreateNeedFactory.getRegions();
+            if(!$sessionStorage.token){
+                $state.go('login');
```
Comment (standemchuk) [inline src/client/app/needs/new_need_register.controller.js]: save the attempt url prior to redirecting user to the login page.
and after a successful login, redirect the user to the page he wanted, not the home page.

File: src/client/app/needs/new_need_step_1.controller.js [hunk 3]
```diff
@@ -71,7 +71,11 @@
         };
         //getting categories object from service
         function activate() {
-            vm.categories = CreateNeedFactory.getCategories();
+            if(!$sessionStorage.token){
+                $state.go('login');
+            }else{
+                vm.categories = CreateNeedFactory.getCategories();
+            }
```
Comment (standemchuk) [inline src/client/app/needs/new_need_step_1.controller.js]: same here

File: src/client/app/offers/new_offer_register.js [hunk 4]
```diff
@@ -46,7 +46,11 @@
         };
 
         function activate() {
-            vm.regions = CreateOfferFactory.getRegions();
+            if(!$sessionStorage.token){
+                $state.go('login');
+            }else{
+                vm.regions = CreateOfferFactory.getRegions();
+            }
```
Comment (standemchuk) [inline src/client/app/offers/new_offer_register.js]: and here

File: src/client/app/offers/new_offer_step1.js [hunk 5]
```diff
@@ -71,7 +71,11 @@
         };
         //getting categories object from service
         function activate() {
-            vm.categories = CreateOfferFactory.getCategories();
+            if(!$sessionStorage.token){
+                $state.go('login');
+            }else{
+                vm.categories = CreateOfferFactory.getCategories();
+            }
```
Comment (standemchuk) [inline src/client/app/offers/new_offer_step1.js]: and here

File: src/client/app/header/app.header.controller.js (hunk 1 above)
Comment (alusbars) [inline src/client/app/header/app.header.controller.js]: Token is not a persistent object, security generates new token once per request to /api/auth and it has ttl for 10 days. After logout reseting your session storage is enough, but it's not very good that cleaned token still will be valid for remaining period of time. I've already created bug for this issue https://jira.softserveinc.com/browse/CHSP-225

**Pull request merged**